Isabelle/Chi_Calculus_Examples/Diamond.thy

```diff
@@ -194,26 +194,13 @@ proof -
     sorry
   then have "a \<triangleright>\<^sup>\<infinity> x. \<Prod> c\<leftarrow>[b]. c \<triangleleft> x \<approx>\<^sub>\<flat> a \<triangleright>\<^sup>\<infinity> x. b \<triangleleft> x"
     by (simp add: basic_weak_multi_receive_preservation)
-  then show ?thesis .
+  then show ?thesis
+    by simp
 qed
 
 lemma singleton_distributor_switch:
   shows "a \<leftrightarrow> b \<parallel> c \<Rightarrow> [a] \<approx>\<^sub>\<flat> a \<leftrightarrow> b \<parallel> c \<rightarrow> b"
-proof -
-  have "a \<leftrightarrow> b \<parallel> c \<Rightarrow> [a] \<approx>\<^sub>\<flat> a \<leftrightarrow> b \<parallel> c \<rightarrow> a"
-    using singleton_distribution sorry
-  also have "a \<leftrightarrow> b \<parallel> c \<rightarrow> a \<approx>\<^sub>\<flat> a \<leftrightarrow> b \<parallel> c \<triangleright>\<^sup>\<infinity> x. (a \<rightarrow> b \<parallel> a \<triangleleft> x)"
-    using context_localization sorry
-  also have "a \<leftrightarrow> b \<parallel> c \<triangleright>\<^sup>\<infinity> x. (a \<rightarrow> b \<parallel> a \<triangleleft> x) \<approx>\<^sub>\<flat> a \<leftrightarrow> b \<parallel> c \<triangleright>\<^sup>\<infinity> x. (a \<leftrightarrow> b \<parallel> a \<triangleleft> x)"
-    using context_localization sorry
-  also have "a \<leftrightarrow> b \<parallel> c \<triangleright>\<^sup>\<infinity> x. (a \<leftrightarrow> b \<parallel> a \<triangleleft> x) \<approx>\<^sub>\<flat> a \<leftrightarrow> b \<parallel> c \<triangleright>\<^sup>\<infinity> x. (a \<leftrightarrow> b \<parallel> b \<triangleleft> x)"
-    using send_channel_switch and basic_weak_multi_receive_preservation sorry
-  also have "a \<leftrightarrow> b \<parallel> c \<triangleright>\<^sup>\<infinity> x. (a \<leftrightarrow> b \<parallel> b \<triangleleft> x) \<approx>\<^sub>\<flat> a \<leftrightarrow> b \<parallel> c \<triangleright>\<^sup>\<infinity> x. (a \<rightarrow> b \<parallel> b \<triangleleft> x)"
-    using context_localization sorry
-  also have "a \<leftrightarrow> b \<parallel> c \<triangleright>\<^sup>\<infinity> x. (a \<rightarrow> b \<parallel> b \<triangleleft> x) \<approx>\<^sub>\<flat> a \<leftrightarrow> b \<parallel> c \<rightarrow> b"
-    using context_localization sorry
-  finally show ?thesis .
-qed
+  using target_switch by blast
```

Comment:
Isn’t `singleton_distributor_switch` the same as `target_switch` now? If yes, we could replace `by blast` with `.`. However, it would actually be better then to drop `singleton_distributor_switch` and use `target_switch` instead of it.

Reply:
Right.